lib/api.ts

```diff
@@ -180,6 +186,24 @@ export async function getContributorBySlug(file: string, detail = false) {
 
   const calendarData = getCalendarData(weightedActivity.activity);
 
+  const isNewContributor = (() => {
+    const { activity } = activityData;
+
+    if (!activity || activity.length === 0) return false;
+
+    const prMergedActivities = activity
+      .filter((act) => act.type === "pr_merged")
+      .sort((a, b) => new Date(a.time).getTime() - new Date(b.time).getTime());
+
+    if (prMergedActivities.length === 0) return false;
+
+    const firstActivityTime = new Date(prMergedActivities[0].time);
+
+    return (
+      firstActivityTime >= lastWeekStart && firstActivityTime <= lastWeekEnd
+    );
```

Comment:
you can reduce the complexity by using `Array.some(...)` instead.

```suggestion
    const aWeekAgo = sub(new Date(), 7 days);
    
    return activity?.some((activity) => {
      if (activity.type !== "pr_merged") return false;
      return isAfter(activity, aWeekAgo));
    });
```

Reply:
I believe this needs some adjustments. It should only count the first PR merged by each contributor, not all PRs merged by them, right @rithviknishad  ?

Reply:
oh yes, my bad